Automattic/wp-calypso
# Reader - Add suggested follows modal

This PR will add a modal that will show a list of recommended follows based on the site the user has just followed.

![follow-1](https://github.com/Automattic/wp-calypso/assets/5560595/baeddf16-29ea-4209-82ec-77da48ca647b)

We will only show this modal on Reader site page. We may also add it to Reader search page.

Project Ref: pe7F0s-O2-p2

### PR highlights

The PR requires the patch at code-D111404. The patch passes back extra site information for each related post, saving having to make an extra `QueryReaderSite` call for each related post, allowing the dialog to render more quickly.

This PR adds a new scope of `SCOPE_SUGGESTED_FOLLOWS` to allow us to pass extra params to the related posts endpoint.

The PR requests the relatedPosts when the full site page loads, **not** when the dialog loads. This is done this way to avoid waiting on the dialog to load while the related posts request is processing.

Another thing of note, the Follow button style is left unchanged for now. The designs have a button style but this would require a change to the followbutton component which is something I think we can come back to.

### Testing
* Apply patch code-D111404 on sandbox
* Sandbox public-api.wordpress.com
* Go to a reader site page like http://calypso.localhost:3000/read/feeds/35154032
* Click the `Follow` button and you should see a dialog like the image below - confirm it works and looks as expected

<img width="514" alt="Screenshot 2023-05-19 at 16 15 16" src="https://github.com/Automattic/wp-calypso/assets/5560595/f0c27726-73e4-4ab5-a3af-6b105c779796">
